flows/index.py

```diff
@@ -888,45 +934,33 @@ async def index_by_s3(
     with cm:
         logger.info("getting S3 object generator")
         documents_generator = s3_obj_generator(s3_prefixes, s3_paths)
-
         documents_batches = iterate_batch(documents_generator, batch_size=batch_size)
 
-        for (
-            documents_batch_num,
-            documents_batch,
-        ) in enumerate(documents_batches, start=1):
-            logger.info(f"processing batch documents #{documents_batch_num}")
-
-            indexing_tasks = [
-                run_partial_updates_of_concepts_for_document_passages_as(
-                    document_importer=document_importer,
-                    as_subflow=as_subflow,
-                    aws_env=aws_env,
-                    cache_bucket=cache_bucket,
-                    concepts_counts_prefix=concepts_counts_prefix,
-                )
-                for document_importer in documents_batch
-            ]
+        flow_name = function_to_flow_name(run_partial_updates_of_concepts_for_batch)
+        deployment_name = generate_deployment_name(flow_name=flow_name, aws_env=aws_env)
 
-            logger.info(
-                f"gathering {batch_size} indexing tasks for batch {documents_batch_num}"
-            )
-            results = await asyncio.gather(*indexing_tasks, return_exceptions=True)
-            logger.info(
-                f"gathered {batch_size} indexing tasks for batch {documents_batch_num}"
+        indexing_tasks = [
+            run_deployment(
+                name=f"{flow_name}/{deployment_name}",
+                parameters={
+                    "documents_batch": documents_batch,
+                    "batch_size": batch_size,
+                    "documents_batch_num": documents_batch_num,
+                    "aws_env": aws_env,
+                    "cache_bucket": cache_bucket,
+                    "concepts_counts_prefix": concepts_counts_prefix,
+                    "as_subflow": as_subflow,
+                },
             )
+            for (
+                documents_batch_num,
+                documents_batch,
+            ) in enumerate(documents_batches, start=1)
+        ]
 
-            for i, result in enumerate(results):
-                # Get the S3 key for the task
-                document_import_id: DocumentImportId = documents_batch[i][0]
-
-                if isinstance(result, Exception):
-                    logger.error(
-                        f"failed to process document `{document_import_id}`: {str(result)}",
-                    )
-                    continue
-
-                logger.info(f"processed batch documents #{documents_batch_num}")
+        logger.info("gathering indexing tasks")
+        _ = await asyncio.gather(*indexing_tasks, return_exceptions=True)
```

Comment:
is it fine for the results of this gather to be discarded? is there any value in logging or retrying failures at this level?

Reply:
Good shout I'll add back in.